metamorph/processor_response_map.go

```diff
@@ -53,25 +53,26 @@ func NewProcessorResponseMap(expiry time.Duration) *ProcessorResponseMap {
 
 func (m *ProcessorResponseMap) logWriter() {
 	dir := path.Dir(m.logFile)
-	_ = os.MkdirAll(dir, 0777)
-
-	f, err := os.OpenFile(m.logFile, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	err := os.MkdirAll(dir, 0750)
+	if err != nil {
+		log.Fatalf("failed to create folder for logging %s", err)
+	}
+	f, err := os.OpenFile(m.logFile, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0600)
 	if err != nil {
-		log.Printf("error opening log file: %s", err.Error())
+		log.Fatalf("error opening log file: %s", err)
```

Comment:
I think `Fatalf()` is not to be used here. Because Fatalf is equivalent to Printf() followed by a call to os.Exit(1).
We shouldn't exit if an error occurs, but just log it.

Regarding the formatted string. We should print the error either like this `"... %s", err.String()` or `"... %v", err` so that the error message will be printed.

Reply:
1. I'm not sure how it would proceed with the rest if there is an error with opening file. 
2. I could not find `err.String` in standard errors package, probably you meant err.Error()?
3. In this particular case "%s" and "%v" can serve same purpose and .Error() method will be called.

Reply:
Oh yes, I meant `err.Error()`. Ok, didn't know that 👍

Reply:
1. Ok, I see that point